code/modules/mob/living/simple_animal/bot/honkbot.dm

```diff
@@ -0,0 +1,380 @@
+			retaliate(H)
+	..()
+
+/mob/living/simple_animal/bot/honkbot/proc/bike_horn() //use bike_horn
+	if (emagged <= 1)
+		if(!client) //check if a player is controlling
+			playsound(src, honksound, 50, 1, -1)
+		else
+			if (!spam_flag)
+				playsound(src, honksound, 50, 1, -1)
+				spam_flag = TRUE //prevent spam
+		sensor_blink()
+		addtimer(CALLBACK(src, .proc/spam_flag_false), cooldowntimehorn)
+
+	else if (emagged == 2) //emagged honkbots will spam short and memorable sounds.
+
+		if (!client)
+			playsound(src, "honkbot_e", 50, 0)
+		else
+			if (!spam_flag)
+				playsound(src, "honkbot_e", 50, 0)
+				spam_flag = TRUE // prevent spam
+		icon_state = "honkbot-e"
+		spawn(30) // keep flashing
+			icon_state = "honkbot[on]"
+		addtimer(CALLBACK(src, .proc/spam_flag_false), cooldowntimehorn)
+
+
+/mob/living/simple_animal/bot/honkbot/proc/honk_attack(mob/living/carbon/C) // horn attack
+	if (!client) //check if a player is controlling
+		playsound(loc, honksound, 50, 1, -1)
+	else
+		if(!spam_flag)
+			playsound(loc, honksound, 50, 1, -1)
+			spam_flag = TRUE // prevent spam
+	sensor_blink()
+	addtimer(CALLBACK(src, .proc/spam_flag_false), cooldowntimehorn)
+
+/mob/living/simple_animal/bot/honkbot/proc/stun_attack(mob/living/carbon/C) // airhorn stun
+	if (!client)
+		playsound(loc, 'sound/items/AirHorn.ogg', 100, 1, -1) //HOOOOOOOOOOOOONK!!
+	else
+		if(!spam_flag)
+			playsound(loc, 'sound/items/AirHorn.ogg', 100, 1, -1) //HEEEEEEEEEEEENK!!
+	sensor_blink()
+	if(spam_flag == 0)
+		if(ishuman(C))
+			C.stuttering = 20
+			C.adjustEarDamage(0, 5) //far less damage than the H.O.N.K.
+			C.Jitter(50)
+			C.Knockdown(60)
+			var/mob/living/carbon/human/H = C
+			if(client) //prevent spam from players..
+				spam_flag = TRUE
+			if (emagged <= 1) //HONK once, then leave
+				threatlevel = H.assess_threat(src)
+				threatlevel -= 6
+				target = oldtarget_name
+			else // you really don't want to hit an emagged honkbot
+				threatlevel = 6 // will never let you go
+			addtimer(CALLBACK(src, .proc/spam_flag_false), cooldowntime)
+
+			add_logs(src,C,"honked")
+
+
+			C.visible_message("<span class='danger'>[src] has honked [C]!</span>",\
+					"<span class='userdanger'>[src] has honked you!</span>")
+
+		else
+			C.stuttering = 20
+			C.Knockdown(80)
+			addtimer(CALLBACK(src, .proc/spam_flag_false), cooldowntime)
+
+
+/mob/living/simple_animal/bot/honkbot/handle_automated_action()
+	if(!..())
+		return
+
+	switch(mode)
+
+		if(BOT_IDLE)		// idle
+
+			walk_to(src,0)
+			look_for_perp()
+			if(!mode && auto_patrol)
+				mode = BOT_START_PATROL
+
+		if(BOT_HUNT)
+
+			// if can't reach perp for long enough, go idle
+			if(frustration >= 5) //gives up easier than beepsky
+				walk_to(src,0)
+				back_to_idle()
+				return
+
+			if(target)		// make sure target exists
+				if(Adjacent(target) && isturf(target.loc))
+
+					if(threatlevel <= 4)
+						honk_attack(target)
+					else
+						if(threatlevel >= 6)
+							spawn(0)
```

Comment:
waitfor =  FALSE on stun attack and moving the anchor resets there should be better.

Reply:
to replace the spawn(0), right? On it.